compiler/ballerina-treegen/src/main/resources/syntax_tree_descriptor.json

```
@@ -4371,6 +4371,22 @@
                     "type": "ExpressionNode"
                 }
             ]
+        },
+        {
+            "name": "TupleMemberDescriptorNode",
+            "base": "TypeDescriptorNode",
```

We extend a node with `TypeDescriptorNode` only if the node itself is a type descriptor. We cannot use this in other places we have type-descriptor.

```suggestion
            "base": "Node",
```

Also, when updating this file, we need to update `api_gen_syntax_tree_descriptor.json` in parallel. Ballerina shell uses that file. We already have issue #35389 to avoid duplication.